projects/parsers/loop_report_parser.py

```diff
@@ -0,0 +1,256 @@
+
+class Sections():
+    LOOP_VERSION = "loop_version"
+    DEVICE_DATA_MANAGER = "device_data_manager"
+    RILEY_LINK_DEVICE = "riley_link_device"
+    CARB_STORE = "carb_store"
+    DOSE_STORE = "dose_store"
+    MINIMED_PUMP_MANAGER = "minimed_pump_manager"
+    OMNIPOD_PUMP_MAANGER ="omnipod_pump_manager"
+    WATCH_DATA_MANAGER ="watch_data_manager"
+    LOOP_DATA_MANAGER ="loop_data_manager"
+
+
+def _split_key_value(line, separator):
+    end_loc = line.find(separator)
+    key = line[0:end_loc]
+    value = line[end_loc + 1:len(line)]
+    return key, value
+
+def parse_loop_report(dataPathAndName):
+    current_section = ""
+    all_sections = {}
+    new_line = False
+
+    with open(dataPathAndName, 'r') as reader:
+        for line in reader:
+            if line.startswith("Generated:"):
+                key, value = _split_key_value(line, ":")
+                generated = {}
+                generated[key] = value
+                all_sections["generated"] = generated
+                new_line = False
+
+            elif line.startswith('Loop'):
+                key, value = _split_key_value(line, ":")
+                loop = {}
+                loop["loop_version"] = key
+                all_sections["loop_version"] = loop
+                new_line = False
+
+            elif line.startswith('## DeviceDataManager'):
+                device_data_manager = {}
+                current_section = "device_data_manager"
+                all_sections["device_data_manager"] = device_data_manager
+                new_line = False
+
+            elif line.startswith('## G5CGMManager'):
+                g5_cgm_manager = {}
+                current_section = "g5_cgm_manager"
+                all_sections["g5_cgm_manager"] = g5_cgm_manager
+                new_line = False
+
+            elif line.startswith('## DexCGMManager'):
+                dex_cgm_manager = {}
+                current_section = "dex_cgm_manager"
+                all_sections["dex_cgm_manager"] = dex_cgm_manager
+                new_line = False
+
+
+            elif line.startswith('## MinimedPumpManager'):
+                minimed_pump_manager = {}
+                current_section = "minimed_pump_manager"
+                all_sections["minimed_pump_manager"] = minimed_pump_manager
+                new_line = False
+
+            elif line.startswith('## RileyLinkPumpManager'):
+                riley_link_pump_manager = {}
+                current_section = "riley_link_pump_manager"
+                all_sections["riley_link_pump_manager"] = riley_link_pump_manager
+                new_line = False
+
+            elif line.startswith('## RileyLinkDeviceManager'):
+                riley_link_device_manager = {}
+                current_section = "riley_link_pump_manager"
+                all_sections["riley_link_pump_manager"] = riley_link_pump_manager
```

Comment:
should this be `riley_link_device_manager` instead of `riley_link_pump_manager `